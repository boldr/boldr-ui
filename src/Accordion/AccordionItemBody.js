/* @flow */
import React from 'react';
import type { Node } from 'react';
import classNames from 'classnames';

type Props = {
  id: string,
  isExpanded: boolean,
  children: Array<Node>,
  className: string,
  hideBodyClassName: string,
  role: string,
};

const defaultProps = {
  isExpanded: false,
  className: 'boldr-accordion__body',
  hideBodyClassName: 'boldr-accordion__body--hidden',
};

const AccordionItemBody = ({
  id,
  isExpanded,
  children,
  className,
  hideBodyClassName,
  role,
}: Props) => {
  const bodyClass = classNames(className, {
    [hideBodyClassName]: !isExpanded,
  });
  const ariaHidden = !isExpanded;
  return (
    <div id={id} className={bodyClass} aria-hidden={ariaHidden} role={role}>
      {children}
    </div>
  );
};

AccordionItemBody.defaultProps = defaultProps;
AccordionItemBody.accordionElementName = 'AccordionItemBody';

export default AccordionItemBody;
