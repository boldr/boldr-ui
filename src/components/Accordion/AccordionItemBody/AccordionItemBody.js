/* @flow */
import React from 'react';
import classNames from 'classnames';
import type { ReactChildren } from '../../../types/react.js.flow';

type Props = {
  id: string,
  expanded: boolean,
  onClick: () => void,
  ariaControls: string,
  children: ReactChildren,
  className: string,
  hideBodyClassName: string,
  role: string,
};

const defaultProps = {
  id: '',
  expanded: false,
  className: 'boldrui-accordion__body',
  hideBodyClassName: 'boldrui-accordion__body--hidden',
  role: '',
};

const AccordionItemBody = ({
  id,
  expanded,
  children,
  className,
  hideBodyClassName,
  role,
}: Props) => {
  const bodyClass = classNames(className, {
    [hideBodyClassName]: !expanded,
  });
  const ariaHidden = !expanded;
  return (
    <div id={id} className={bodyClass} aria-hidden={ariaHidden} role={role}>
      {children}
    </div>
  );
};

AccordionItemBody.defaultProps = defaultProps;
AccordionItemBody.accordionElementName = 'AccordionItemBody';

export default AccordionItemBody;
