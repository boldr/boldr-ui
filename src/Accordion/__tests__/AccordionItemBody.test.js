import React from 'react';
import renderer from 'react-test-renderer';

import AccordionItemBody from '../AccordionItemBody';

describe('AccordionItemBody', () => {
  it('renders correctly with min params', () => {
    const tree = renderer
      .create(
        <AccordionItemBody>
          <div>Fake body</div>
        </AccordionItemBody>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with different className', () => {
    const tree = renderer
      .create(
        <AccordionItemBody className="testCSSClass">
          <div>Fake body</div>
        </AccordionItemBody>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with prefixClass', () => {
    const tree = renderer
      .create(
        <AccordionItemBody hideBodyClassName="testCSSClass--hidden">
          <div>Fake body</div>
        </AccordionItemBody>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with expanded', () => {
    const tree = renderer
      .create(
        <AccordionItemBody isExpanded>
          <div>Fake body</div>
        </AccordionItemBody>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
