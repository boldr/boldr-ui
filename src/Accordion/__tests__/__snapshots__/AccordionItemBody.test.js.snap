// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionItemBody renders correctly with different className 1`] = `
<div
  aria-hidden={true}
  className="testCSSClass boldr-accordion__body--hidden"
  id={undefined}
  role={undefined}
>
  <div>
    Fake body
  </div>
</div>
`;

exports[`AccordionItemBody renders correctly with expanded 1`] = `
<div
  aria-hidden={false}
  className="boldr-accordion__body"
  id={undefined}
  role={undefined}
>
  <div>
    Fake body
  </div>
</div>
`;

exports[`AccordionItemBody renders correctly with min params 1`] = `
<div
  aria-hidden={true}
  className="boldr-accordion__body boldr-accordion__body--hidden"
  id={undefined}
  role={undefined}
>
  <div>
    Fake body
  </div>
</div>
`;

exports[`AccordionItemBody renders correctly with prefixClass 1`] = `
<div
  aria-hidden={true}
  className="boldr-accordion__body testCSSClass--hidden"
  id={undefined}
  role={undefined}
>
  <div>
    Fake body
  </div>
</div>
`;
